pkg/testgridanalysis/testgridconversion/to_raw_data.go

```diff
@@ -227,45 +236,50 @@ func processTestToJobRunResults(jobResult testgridanalysisapi.RawJobResult, job
 		testName = testgridanalysisapi.OperatorFinalHealthPrefix + " " + operatorName
 	}
 
-	addTestResult(jobResult.TestResults, testName, passed, failed, flaked)
+	addTestResult(jobResult.TestResults, testgridanalysisapi.RawTestResult{
+		Name:      testName,
+		Successes: passed,
+		Failures:  failed,
+		Flakes:    flaked,
+	})
 
 	return
 }
 
-func processTest(rawJobResults testgridanalysisapi.RawData, job testgridv1.JobDetails, test testgridv1.Test, startCol, endCol int) {
+func processTest(rawJobResults testgridanalysisapi.RawData, job testgridv1.JobDetails, test testgridv1.Test, lookbackCols lookback) {
 	// strip out tests that don't have predictive or diagnostic value
 	// we have to know about overall to be able to set the global success or failure.
 	// we have to know about container setup to be able to set infra failures
 	// TODO stop doing this so we can avoid any filtering. We can filter when preparing to create the data for display
-	if test.Name != "Overall" && !testidentification.IsSetupContainerEquivalent(test.Name) && ignoreTestRegex.MatchString(test.Name) {
+	if test.Name != overall && !testidentification.IsSetupContainerEquivalent(test.Name) && ignoreTestRegex.MatchString(test.Name) {
 		return
 	}
 
 	jobResult, ok := rawJobResults.JobResults[job.Name]
 	if !ok {
 		jobResult = testgridanalysisapi.RawJobResult{
 			JobName:        job.Name,
-			TestGridJobUrl: job.TestGridUrl,
+			TestGridJobURL: job.TestGridURL,
 			JobRunResults:  map[string]testgridanalysisapi.RawJobRunResult{},
 			TestResults:    map[string]testgridanalysisapi.RawTestResult{},
 		}
 	}
 
-	processTestToJobRunResults(jobResult, job, test, startCol, endCol)
+	processTestToJobRunResults(jobResult, job, test, lookbackCols)
 
 	// we have mutated, so assign back to our intermediate value
 	rawJobResults.JobResults[job.Name] = jobResult
 }
 
-func addTestResult(testResults map[string]testgridanalysisapi.RawTestResult, testName string, passed, failed, flaked int) {
-	result, ok := testResults[testName]
-	if !ok {
-		result = testgridanalysisapi.RawTestResult{}
+func addTestResult(testResults map[string]testgridanalysisapi.RawTestResult, testResult testgridanalysisapi.RawTestResult) {
+	result, ok := testResults[testResult.Name]
+	if ok {
+		result.Successes += testResult.Successes
+		result.Failures += testResult.Failures
+		result.Flakes += testResult.Flakes
+	} else {
+		result = testResult
 	}
-	result.Name = testName
-	result.Successes += passed
-	result.Failures += failed
-	result.Flakes += flaked
 
-	testResults[testName] = result
```

Comment:
i'm struggling to see this as a strict improvement.

previously the code always set the result.* fields based on the testResult.* fields.

now there's an if/else that effectively does the same thing, but in two different ways depending on whether you had an existing value or not.

Reply:
This one was entirely on me. After sleeping on it, I agree.